### PR TITLE
FIX-#0000: fix xgboost tests with ray>2.6.0

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

Fixes xgboost tests on master branch.

Reference: https://github.com/modin-project/modin/actions/runs/5694815741/job/15436853389

```python
...
FAILED modin/experimental/xgboost/test/test_xgboost.py::test_xgb_with_regression_datasets[load_diabetes-3-DataFrame] - ZeroDivisionError: division by zero
FAILED modin/experimental/xgboost/test/test_xgboost.py::test_xgb_with_regression_datasets[load_diabetes-3-Series] - ZeroDivisionError: division by zero
FAILED modin/experimental/xgboost/test/test_xgboost.py::test_invalid_input - ZeroDivisionError: division by zero
```

Here one can verify that the problem has been fixed: https://github.com/modin-project/modin/actions/runs/5701265682/job/15451811834

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [ ] Resolves #? 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Comment by anmyachev** [comment]
Blocked on #6428

File: modin/experimental/xgboost/xgboost_ray.py [hunk 1]
```diff
@@ -268,10 +268,17 @@ def create_actors(num_actors):
...
+    # {'object_store_memory': 1036438732.0, 'memory': 2072877467.0, 'node:127.0.0.1': 1.0, 'CPU': 8.0, 'node:__internal_head__': 1.0}
     node_ips = [
-        key for key in ray.cluster_resources().keys() if key.startswith("node:")
+        key
+        for key in ray.cluster_resources().keys()
+        if key.startswith("node:") and "__internal_head__" not in key
     ]
-    num_actors_per_node = num_actors // len(node_ips)
+
+    num_actors_per_node = max(num_actors // len(node_ips), 1)
```
Comment (RehanSD) [inline modin/experimental/xgboost/xgboost_ray.py:281]: Why is the max added here?

**Review by RehanSD — APPROVED** [review]
Left a quick q, but LGTM otherwise!

File: modin/experimental/xgboost/xgboost_ray.py (hunk 1 above)
Comment (anmyachev) [reply modin/experimental/xgboost/xgboost_ray.py:281]: There is a case that seems problematic to me:
if num_actors == 1 and len(node_ips) > 1 then num_actors_per_node == 0. This also results in the following error: `ZeroDivisionError: division by zero`.

In this case, we can either give an exception, or take 1 actor at a minimum.

**Comment by anmyachev** [comment]
Blocked on #6429

**Review by YarShev — APPROVED** [review]

**Comment by YarShev** [comment]
@modin-project/modin-xgboost, we need your approve

**Review by mvashishtha — APPROVED** [review]

File: modin/experimental/xgboost/xgboost_ray.py [hunk 2]
```diff
@@ -268,10 +268,17 @@ def create_actors(num_actors):
         List of pairs (ip, actor).
     """
     num_cpus_per_actor = _get_cpus_per_actor(num_actors)
+    # starting from ray 2.6 there is a new field: 'node:__internal_head__'
+    # example:
+    # >>> ray.cluster_resources()
+    # {'object_store_memory': 1036438732.0, 'memory': 2072877467.0, 'node:127.0.0.1': 1.0, 'CPU': 8.0, 'node:__internal_head__': 1.0}
```
Comment (YarShev) [inline modin/experimental/xgboost/xgboost_ray.py:274]: @rkooo567, I wonder if the key `'node:__internal_head__'` is kind of internal stuff? Why has Ray started exposing this?

Comment (rkooo567) [reply modin/experimental/xgboost/xgboost_ray.py:274]: Yeah this does seem internal. I assume it is to allow to schedule on a head node. https://github.com/ray-project/ray/commit/0ade315bec8265274c1d2c933d99d3b335c46620

Comment (YarShev) [reply modin/experimental/xgboost/xgboost_ray.py:274]: @rkooo567, should this be hidden from the user as it seems internal?

Comment (rkooo567) [reply modin/experimental/xgboost/xgboost_ray.py:274]: I think cluster_resources is the developerAPI, and we don't hide resources to end users from this API. I believe it is hidden from ray status

Comment (YarShev) [reply modin/experimental/xgboost/xgboost_ray.py:274]: Got it, thanks!

**Pull request merged**